drivers/ieee802154/ieee802154_mcxw.c

```diff
@@ -540,10 +549,11 @@ static int mcxw_tx(const struct device *dev, enum ieee802154_tx_mode mode, struc
 #if defined(CONFIG_NET_PKT_TXTIME)
 	case IEEE802154_TX_MODE_TXTIME:
 	case IEEE802154_TX_MODE_TXTIME_CCA:
-		mcxw_radio->tx_frame.tx_delay = net_pkt_timestamp_ns(pkt);
+		mcxw_radio->tx_frame.tx_delay -= IEEE802154_SHR_DURATION_US;
```

Comment:
This is not correct,
`tx_delay` is in nanoseconds while SHR duration is in us.

See https://github.com/zephyrproject-rtos/zephyr/pull/100389/files#r2601444923